test/unit/StakingPool/depositTo.js

```diff
@@ -0,0 +1,774 @@
+      {
+        amount,
+        trancheId: firstActiveTrancheId,
+        tokenId,
+        destination,
+      },
+    ]);
+
+    const firstDepositData = await stakingPool.deposits(depositNftId, firstActiveTrancheId);
+
+    const newStakeShares = await estimateStakeShares({ amount, stakingPool });
+
+    // deposit to the same tokenId
+    await stakingPool.connect(user).depositTo([
+      {
+        amount,
+        trancheId: firstActiveTrancheId,
+        tokenId: depositNftId,
+        destination,
+      },
+    ]);
+
+    const updatedDepositData = await stakingPool.deposits(depositNftId, firstActiveTrancheId);
+    const newRewardShares = await getNewRewardShares({
+      stakingPool,
+      initialStakeShares: firstDepositData.stakeShares,
+      stakeSharesIncrease: newStakeShares,
+      initialTrancheId: firstActiveTrancheId,
+      newTrancheId: firstActiveTrancheId,
+    });
+    expect(updatedDepositData.pendingRewards).to.equal(0);
+    expect(updatedDepositData.lastAccNxmPerRewardShare).to.equal(0);
+    expect(updatedDepositData.stakeShares).to.equal(firstDepositData.stakeShares.add(newStakeShares));
+    expect(updatedDepositData.rewardsShares).to.equal(firstDepositData.rewardsShares.add(newRewardShares));
+  });
+
+  it('updates deposit pendingRewards and lastAccNxmPerRewardShare', async function () {
+    const { stakingPool } = this;
+    const {
+      members: [user],
+    } = this.accounts;
+
+    const { amount, tokenId, destination, depositNftId } = depositToFixture;
+
+    const { firstActiveTrancheId } = await getTranches();
+
+    // first deposit
+    await stakingPool.connect(user).depositTo([
+      {
+        amount,
+        trancheId: firstActiveTrancheId,
+        tokenId,
+        destination,
+      },
+    ]);
+
+    const depositData = await stakingPool.deposits(depositNftId, firstActiveTrancheId);
+
+    expect(depositData.lastAccNxmPerRewardShare).to.equal(0);
+    expect(depositData.pendingRewards).to.equal(0);
+
+    // Generate rewards
+    const coverRequest = {
+      coverId: 0,
+      productId: 0,
+      amount: parseEther('100'),
+      period: 3600 * 24 * 30, // 30 days
+      gracePeriod: 3600 * 24 * 30,
+      globalCapacityRatio: 20000,
+      capacityReductionRatio: 0,
+      rewardRatio: 10000,
+    };
+    await stakingPool.connect(this.coverSigner).allocateStake(coverRequest);
+
+    await increaseTime(
+      20 * // days
+        24 * // hours
+        60 * // minutes
+        60, // seconds
+    );
+
+    // Second deposit
+    await stakingPool.connect(user).depositTo([
+      {
+        amount,
+        trancheId: firstActiveTrancheId,
+        tokenId: depositNftId,
+        destination,
+      },
+    ]);
+
+    const secondDepositData = await stakingPool.deposits(depositNftId, firstActiveTrancheId);
+    const secondAccNxmPerRewardsShare = await stakingPool.accNxmPerRewardsShare();
+
+    expect(secondAccNxmPerRewardsShare).to.not.equal(0);
```

Comment:
we want to test it's the right value not just different from 0 we should compute an exact value here

Reply:
The expects were confusing, here we are validating `secondDepositData.lastAccNxmPerRewardShare` got updated. In the next line, it is checking the correct value. The intention was to validate that it got assigned the correct value (the global accNxmPerRewardsShare) and that the correct value is not 0. I didn't want to over complex the test with the specific calculation of accNxmPerRewardsShare as its value depends from the calculation in `updateTranches()` which eventually will be properly tested and we can validate the calculations there.

Updated the code to reflect that better here [Update deposit lastAccNxmPerRewardShare expects](https://github.com/NexusMutual/smart-contracts/pull/495/commits/9811ccc60543ae5521b127dc8ab7371f25808d4f)